pkg/agent/route/route_linux.go

```diff
@@ -94,13 +95,18 @@ func (c *Client) Initialize(nodeConfig *config.NodeConfig, done func()) error {
 	c.nodeConfig = nodeConfig
 
 	// Sets up the ipset that will be used in iptables.
-	if err := c.initIPSet(); err != nil {
+	if err := c.syncIPSet(); err != nil {
 		return fmt.Errorf("failed to initialize ipset: %v", err)
 	}
 
 	// Sets up the iptables infrastructure required to route packets in host network.
 	// It's called in a goroutine because xtables lock may not be acquired immediately.
-	go c.initIPTablesOnce(done)
+	go func() {
+		defer done()
+		var backoffTime = 2 * time.Second
+		c.syncIPTablesWithRetry(backoffTime)
```

Comment:
The routine that initializes the iptables once is simple, I think we either keep all of its logic in one function as it was or keep them in one anonymous without extracting part of it to another function, given that non function is reused or unit tested.

Reply:
Sure @tnqn , I think keeping it anonymous should do, and I'll keep the the xtables contention issue highlighted via comments, as to the reason we want to retry while initialisation. Should need be later, we can extract this into a named method.

Reply:
I think L103 already has the reason why retry is needed.